src/Components/CommentComponent/Comment.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from "react"
+import createComment from "../../helpers/createComment"
+import getAllComments from "../../helpers/getAllComments"
+
+export default class Comment extends Component {
+  state = {
+    comment_arr: [],
+    comment: "",
+    loading: true,
+  }
+
+  commentInputHandler = (e) => {
+    this.setState({ comment: e.target.value })
+  }
+
+  createCommentHandler = async (e) => {
+    const comment = {
+      comment: this.state.comment,
+      createdAt: new Date().getTime(),
+      postID: this.props.postID,
+    }
+    const commentId = await createComment(this.props.postID, comment)
+    const newComment = {
+      commentId: commentId,
+      ...comment,
+    }
+
+    this.setState({
+      comment_arr: [newComment, ...this.state.comment_arr],
+      comment: "",
+    })
+  }
+
+  async componentDidMount() {
+    const arr = await getAllComments(this.props.postID)
+    this.setState({
+      comment_arr: [...arr, ...this.state.comment_arr],
```

Comment:
Why spreading `...this.state.comment_arr`? On mount it's going to be an empty arr.

Reply:
My bad sir.

Reply:
Thought this was event handler.